src/player_display.cpp

```diff
@@ -765,460 +878,560 @@ Strength - 4;    Dexterity - 4;    Intelligence - 4;    Perception - 4" ) );
+                       player.get_dodge(), level.level(), exercise < 0 ? 0 : exercise );
+        } else {
+            mvwprintz( w_skills, int( 1 + i - min ), 19, cstatus, "%-2d(%2d%%)", level.level(),
+                       ( exercise <  0 ? 0 : exercise ) );
+        }
+    }
 
-            case 6: // Effects tab
-                mvwprintz( w_effects, 0, 0, h_light_gray, header_spaces.c_str() );
-                center_print( w_effects, 0, h_light_gray, title_EFFECTS );
-                half_y = effect_win_size_y / 2;
-                if( line <= half_y ) {
-                    min = 0;
-                    max = effect_win_size_y;
-                    if( effect_name.size() < max ) {
-                        max = effect_name.size();
-                    }
-                } else if( line >= effect_name.size() - half_y ) {
-                    min = ( effect_name.size() < effect_win_size_y ? 0 : effect_name.size() - effect_win_size_y );
-                    max = effect_name.size();
-                } else {
-                    min = line - half_y;
-                    max = line - half_y + effect_win_size_y;
-                    if( effect_name.size() < max ) {
-                        max = effect_name.size();
-                    }
-                }
-
-                for( size_t i = min; i < max; i++ ) {
-                    trim_and_print( w_effects, int( 1 + i - min ), 0, getmaxx( w_effects ) - 1,
-                                    i == line ? h_light_gray : c_light_gray, effect_name[i] );
-                }
-                if( line < effect_text.size() ) {
-                    fold_and_print( w_info, 0, 1, FULL_SCREEN_WIDTH - 2, c_magenta, effect_text[line] );
-                }
-                wrefresh( w_effects );
-                wrefresh( w_info );
-
-                action = ctxt.handle_input();
-                if( action == "DOWN" ) {
-                    if( line < effect_name.size() - 1 ) {
-                        line++;
-                    }
-                    break;
-                } else if( action == "UP" ) {
-                    if( line > 0 ) {
-                        line--;
-                    }
-                } else if( action == "NEXT_TAB" ) {
-                    mvwprintz( w_effects, 0, 0, c_light_gray, header_spaces.c_str() );
-                    center_print( w_effects, 0, c_light_gray, title_EFFECTS );
-                    for( size_t i = 0; i < effect_name.size() && i < 7; i++ ) {
-                        mvwprintz( w_effects, int( i + 1 ), 0, c_light_gray, effect_name[i] );
-                    }
-                    wrefresh( w_effects );
-                    line = 0;
-                    curtab = 1;
-                } else if( action == "QUIT" ) {
-                    done = true;
-                }
-                break;
+    draw_scrollbar( w_skills, line, skill_win_size_y, int( skillslist.size() ), 1 );
```

Comment:
If you absolutely must cast a value, please use C++ casts, not C-style casts, e.g. `static_cast<int>( ... )`. The cast here is not needed, the value will be converted automatically.